### DISCUSSION TITLE
Should Strings returned from a Controller be wrapped in quotes when the Content-Type is application/json?

### DISCUSSION THREAD

**Discussion by mjohnston-il** [micronaut-projects/micronaut-core#6615] [General]
When I define a controller endpoint that returns a String object (or HttpResponse<String>) and the @Produces("application/json") annotation is present (or omitted completely - it's the default), the response that is written to the client is not a valid JSON response because the String isn't (necessarily) a quoted String.  Is this something that Micronaut should handle automatically?  A non-String object is automatically serialized to a JSON string, why wouldn't we wrap a String response with quotes to make it a valid JSON string?

Thoughts?

```
@Get("/echo")
@Produces(MediaType.APPLICATION_JSON)
fun echo(@QueryValue("ping")ping: String) : String {
    return ping
}

**Comment by yawkat** [comment]
The reason for the special behavior for `String` is that we want to allow users to return "raw" json without using our serialization logic. If you want to return a json scalar string, you could return a `JsonNode` instead.

**Reply by mjohnston-il** [reply]
I can understand that but it seems ... unintuitive.  I would assume that using a different serializer and outputting already formatted JSON is a fairly uncommon case.  Couldn't that be handled with a marker annotation telling the system to not escape the String?  I realize that for backwards compatibility reasons it might be too late to make a change here regardless.

**Reply by yawkat** [reply]
It might be possible. But you are right in that it wouldn't be backwards-compatible.
